assets/js/blocks/featured-product/block.js

```diff
@@ -125,21 +185,40 @@ class FeaturedProduct extends Component {
 	}
 
 	render() {
-		const { attributes, setAttributes } = this.props;
-		const { editMode } = attributes;
+		const { attributes, setAttributes, overlayColor } = this.props;
+		const {
+			contentAlign,
+			dimRatio,
+			editMode,
+			showDesc,
+			showPrice,
+		} = attributes;
 		const { loaded, product } = this.state;
-		const classes = [ 'wc-block-featured-product' ];
-		if ( ! product ) {
-			if ( ! loaded ) {
-				classes.push( 'is-loading' );
-			} else {
-				classes.push( 'is-not-found' );
-			}
+		const classes = classnames(
+			'wc-block-featured-product',
+			{
+				'is-loading': ! product && ! loaded,
+				'is-not-found': ! product && loaded,
+				'has-background-dim': dimRatio !== 0,
+			},
+			dimRatioToClass( dimRatio ),
+			contentAlign !== 'center' && `has-${ contentAlign }-content`
+		);
+
+		const style = !! product ? backgroundImageStyles( product ) : {};
```

Comment:
It's not necessary here, but I've had unexpected behavior with not explicitly casting to boolean in the render output (it can end up rendering `0` in cases I never fully investigated), so this kind of just became a habit.